libmamba/CMakeLists.txt

```diff
@@ -264,6 +264,22 @@ set(
     ${LIBMAMBA_SOURCE_DIR}/api/shell.cpp
     ${LIBMAMBA_SOURCE_DIR}/api/update.cpp
 )
+# TODO: remove when switch to C++20
+if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
+    # This file uses capturing structured bindings, which was fixed in C++20
+    set_source_files_properties(
+        ${LIBMAMBA_SOURCE_DIR}/download/mirror_impl.cpp
+        PROPERTIES COMPILE_FLAGS -Wno-c++20-extensions
+    )
+endif()
+
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    # GCC reports dangling reference when using std::invoke with data members
+    set_source_files_properties(
+        ${LIBMAMBA_SOURCE_DIR}/solver/problems_graph.cpp
+        PROPERTIES COMPILE_FLAGS -Wno-error=dangling-reference
```

Comment:
This might be useful warning/error, so I'm turning it back to being warning for this file.

Reply:
Is there any way to make that change as part of the source code instead of here (like disabling warnings)?

Reply:
Yes, fortunately it happens in functions located near each other, so we can only write pragmas once.
Thanks!